bin/weewx/drivers/vantage.py

```diff
@@ -1110,14 +1212,27 @@ def hardware_name(self):
     def archive_interval(self):
         return self.archive_interval_
     
+    def determine_hardware(self):
```

Comment:
Why the decision to break out determining the hardware type and give it its own retry loop?